streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractKTableKTableJoinValueGetterSupplier.java

```diff
@@ -30,16 +31,10 @@ public AbstractKTableKTableJoinValueGetterSupplier(final KTableValueGetterSuppli
 
     @Override
     public String[] storeNames() {
-        final String[] storeNames1 = valueGetterSupplier1.storeNames();
-        final String[] storeNames2 = valueGetterSupplier2.storeNames();
-        final ArrayList<String> stores = new ArrayList<>(storeNames1.length + storeNames2.length);
-        for (final String storeName : storeNames1) {
-            stores.add(storeName);
-        }
-        for (final String storeName : storeNames2) {
-            stores.add(storeName);
-        }
-        return stores.toArray(new String[stores.size()]);
+        final ArrayList<String> stores = new ArrayList<>();
+        stores.addAll(Arrays.asList(valueGetterSupplier1.storeNames()));
+        stores.addAll(Arrays.asList(valueGetterSupplier2.storeNames()));
+        return stores.toArray(new String[0]);
```

Comment:
I am not sure, if I like this "array to list to array" thing... It's not a critical code path, but it's quite some boilerplate  (maybe still better than original code, but not much better IMHO).

Maybe `stores.size()` instead of `0` for `new String[.]` ?